File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        stack = []
        
        for c in s:
            stack.append(c)
            while len(stack) >= 2 and (stack[-2] + stack[-1] == "AB" or stack[-2] + stack[-1] == "CD"):
                stack.pop()
                stack.pop()
                
        return len(stack)

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        string minStr = "";
        
        for (const auto& c : s){
            minStr.push_back(c);
            if (minStr.size() >= 2){
                string subs = minStr.substr(minStr.size() - 2, 2);
                if (subs == "AB" || subs == "CD"){
                    minStr.pop_back();
                    minStr.pop_back();
                }
            }       
        }
        
        return minStr.size();
    }
};